server/src/bots/telegram/services/registry/withSingleParameterAfterCommand.ts

```diff
@@ -12,37 +11,39 @@ import { LogCategory } from '../../../../models/constants';
  * which will be an parameter following after command
  */
 export const withSingleParameterAfterCommand = (
-    context: MessageHandlerRegistry,
-    handlerFn: CallBackQueryHandlerWithCommandArgument
+    handlerFn: CallBackQueryHandlerWithCommandArgument,
 ): CallBackQueryHandlerWithCommandArgument => {
-    return (
-        bot: TelegramBot,
-        message: TelegramBot.Message,
-        chatId: number,
-        ikCbData?: string
-    ): Promise<TelegramBot.Message> => {
+    return ({
+                bot,
+                message,
+                chatId,
+                user,
+                messageHandlerRegistry,
+                commandParameter,
+            }: CallBackQueryParameters): Promise<TelegramBot.Message> => {
         try {
             const userEnteredArgumentAfterCommand: string = getParameterAfterCommandFromMessage(
-                context.singleParameterAfterCommands,
-                (ikCbData ?? message.text).toLocaleLowerCase()
+                messageHandlerRegistry.singleParameterAfterCommands,
+                (commandParameter ?? message.text).toLocaleLowerCase(),
             );
 
-            return handlerFn.call(
-                context,
+            return handlerFn.call(messageHandlerRegistry, {
                 bot,
                 message,
                 chatId,
-                userEnteredArgumentAfterCommand
-            );
+                user,
+                messageHandlerRegistry,
+                commandParameter: userEnteredArgumentAfterCommand,
+            });
         } catch (err) {
             logger.error(
                 `Error happend inside withSingleParameterAfterCommand() for ${chatId} with message: ${message.text} and ikCbData: ${ikCbData}`,
                 err,
                 LogCategory.Command,
-                chatId
+                chatId,
             );
 
-            return noResponse(bot, message, chatId);
+            return noResponse({ bot, message, chatId, user, messageHandlerRegistry });
```

Comment:
why do we send whole registry to response layer?

Reply:
will try to remove